### DISCUSSION TITLE
Configure sending certificate chain as a client

### DISCUSSION THREAD

**Discussion by alexandr-osprey** [dotnet/yarp#1427] [Q&A] [answered]
Hello, I am configuring YARP with TLS 1.2. to connect to nginx. nginx has my root certificate which it's supposed to use to validate my chain with. The problem is YARP is not sending a full chain, rather the final certificate. What am I doing wrong?
![image](https://user-images.githubusercontent.com/12857502/144646665-8a5cba9f-df2f-4754-bcc1-db41c0232051.png)

nginx has error 
"client SSL certificate verify error: (21:unable to verify the first certificate) while reading client request headers"

I used wireshark to investigate, here I see that only final certificate is sent, despite 3 of them setup earlier
![image](https://user-images.githubusercontent.com/12857502/144646806-5f9eecfb-f675-4ed4-bb7c-553b6e80eb90.png)

while nginx sends the whole chain
![image](https://user-images.githubusercontent.com/12857502/144646934-0783c4eb-38ed-4d15-90b5-1a33d9370ea8.png)

both client (yarp) and server (nginx) have the same chain

**Comment by karelz** [comment]
I believe that it is normal for `HttpClient` to send only final certificate and not the full chain. I also don't think there is way to configure it otherwise.
@wfurt can you please confirm or correct my claims here?

**Comment by wfurt** [comment, marked as answer]
HttPCLient/SslStream _will_ send the intermediates if they are available. What platform are you running on @alexandr-osprey? On Windows, you have to put the intermediates into "CA" system store.

**Reply by alexandr-osprey** [reply]
Centos 7, net core 3.1
I've managed to get it working by using ~SSL_CERT_PATH~ SSL_CERT_FILE variable pointing to pem chain, as found out in https://github.com/dotnet/runtime/issues/26323#issuecomment-393911780

now final and intermediate certs (without root) are sent, which seem enough for my case

**Comment by wfurt** [comment]
one more note that the .NET _should_ try to get the CAs  if distribution points are in the certificates. That of course depends on network connectivity. (and SSL_CERT_PATH or adding it to Ca store does not)
